### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by dbrattli**
- Add PUT and OPTIONS http methods for convenience.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -47,26 +47,84 @@ type HttpHandler<'err> = HttpHandler<HttpResponseMessage, 'err>
...
+pipeline. Each handler has the full `Context` at its disposal and can decide whether it wants to fail the request or
+continue the request by passing on a new context to the "next" handler.
 
-The more complex way to think about a `HttpHandler` is that there are in fact 3 different ways it may process the
-request:
+1. Call the next handler `NextFunc` with a result value (`'a`), and return (`return!`) what the next handler is
+   returning. Here you have the option to eliding the await by just synchronously return (`return`) the `Task` returned by the
+   `next` function.
+2. Return an `Error`result to short circuit the processing and fail the request.
+3. It is technically possible to also return `Ok` to short circuit the processing, but this is not something you would normally would do.
```
Comment (tengelskar) [inline README.md]: "this is not something you would normally would do." -> "but this is not something you would normally do."

Comment (tengelskar) [inline README.md]:
```suggestion
3. It is technically possible to also return `Ok` to short circuit the processing, but this is not something you would normally do.
```

Comment (dbrattli) [reply README.md]: Thanks, fixed!

File: README.md [hunk 2]
```diff
@@ -343,9 +438,9 @@ The currently defined Metrics are:
 
 Labels are currently not set but are added for future use, e.g setting the error code for fetch errors etc.
 
-## Creating Custom Handlers
+## Extending Oryx
 
-It's easy to extend Oryx with your own HTTP handlers.
+It's easy to extend Oryx with your own context builders and HTTP handlers. Everything is functions so you
```
Comment (tengelskar) [inline README.md]: "Everything is functions so you"
Missing something here?

File: README.md (hunk 1 above)
Comment (dbrattli) [reply README.md]: Fixed

File: README.md (hunk 2 above)
Comment (dbrattli) [reply README.md]: Fixed

**Pull request merged**